lute/process/src/process.cpp

```diff
@@ -474,9 +485,36 @@ int cwd(lua_State* L)
     return 1;
 };
 
+std::optional<std::string> getExecPath(std::string* error)
+{
+    // Executable path is not expected to change during process lifetime, so we
+    // can safely cache it after the first retrieval.
+    static std::optional<std::string> cachedPath = std::nullopt;
+    if (cachedPath)
+        return *cachedPath;
+
+    char buf[LUTE_PATH_MAX];
+    size_t len = sizeof(buf);
+
+    if (int status = uv_exepath(buf, &len); status < 0)
```

Comment:
Does uv_exepath specify the would-be length if it fails? If so, can we remove the predefined 8kb max and just use a std::vector and resize it if need be?